### ISSUE TITLE
`doc_markdown` lint should retreat when citing

### ISSUE THREAD

**Issue by Enyium** [rust-lang/rust-clippy#10262]
### Summary

When citing a source in your doc comment, the source may not be written properly. But as you should know, citations may not be changed without mentioning it. And mentioning changes like "emphasis mine", or whatever you'd write when adding backticks and parantheses, can lead to bloat that you may want to prevent.

So, getting `doc_markdown` warnings inside citations isn't desirable and should be prevented on Clippy's side. Finding out what text is inside `"` or `'` may not produce results accurate enough (the latter also being used for apostrophes). So, maybe Clippy should only pay attention to technical markup for this purpose, namely `<q>...</q>` tags and `>` at the start of lines.

### Lint Name

doc_markdown

### Reproducer

**I tried this code:**

```rust
/// AviSynth+ documentation: "If several scripts are open at once and more than one of them loads this plugin, AvisynthPluginInit3 may be called more than once with different IScriptEnvironments." (<https://avisynthplus.readthedocs.io/en/latest/avisynthdoc/FilterSDK/InvertNeg.html>)
pub type InitFnTest1 = fn(foo: i32) -> Result<(), String>;

/// AviSynth+ documentation: <q>If several scripts are open at once and more than one of them loads this plugin, AvisynthPluginInit3 may be called more than once with different IScriptEnvironments.</q> (<https://avisynthplus.readthedocs.io/en/latest/avisynthdoc/FilterSDK/InvertNeg.html>)
pub type InitFnTest2 = fn(foo: i32) -> Result<(), String>;

/// AviSynth+ documentation:
///
/// > If several scripts are open at once and more than one of them loads this plugin, AvisynthPluginInit3 may be called more than once with different IScriptEnvironments.
/// >
/// > &mdash;<https://avisynthplus.readthedocs.io/en/latest/avisynthdoc/FilterSDK/InvertNeg.html>
pub type InitFnTest3 = fn(foo: i32) -> Result<(), String>;
```

**I saw this happen:**

```
warning: item in documentation is missing backticks
   --> [...].rs:258:178
    |
258 | ...hPluginInit3 may be called more than once with different IScriptEnvironments.</q> (<https://avisynthplus.readthedocs.io/en/latest/avis...
    |                                                             ^^^^^^^^^^^^^^^^^^^
    |
    = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#doc_markdown
help: try
    |
258 | /// AviSynth+ documentation: <q>If several scripts are open at once and more than one of them loads this plugin, AvisynthPluginInit3 may be called more than once with different `IScriptEnvironments`.</q> (<https://avisynthplus.readthedocs.io/en/latest/avisynthdoc/FilterSDK/InvertNeg.html>)
    |                                                                                                                                                                                  ~~~~~~~~~~~~~~~~~~~~~
```

"AvisynthPluginInit3" and "IScriptEnvironments" have yellow wavy underlining. Written properly, they would need to be "`AvisynthPluginInit3()`" and "`IScriptEnvironment`s"; but this isn't to be changed, as I described above. In light of this, recommending "`IScriptEnvironments`" is especially wrong.

**I expected to see this happen:**

There should be no yellow wavy underlining inside `<q>` and `>` quotes.

### Version

```text
rustc 1.67.0 (fc594f156 2023-01-24)
binary: rustc
commit-hash: fc594f15669680fa70d255faec3ca3fb507c3405
commit-date: 2023-01-24
host: x86_64-pc-windows-msvc
release: 1.67.0
LLVM version: 15.0.6
```
